components/org.wso2.carbon.identity.data.publisher.authentication.audit/src/main/java/org/wso2/carbon/identity/data/publisher/authentication/audit/AuthenticationAuditLoggingHandler.java

```diff
@@ -0,0 +1,219 @@
+                + "\",\"" + "AuthenticatedUserTenantDomain" + "\" : \"" + authenticationData.getTenantDomain()
+                + "\",\"" + "ServiceProviderName" + "\" : \"" + authenticationData.getServiceProvider()
+                + "\",\"" + "RequestType" + "\" : \"" + authenticationData.getInboundProtocol()
+                + "\",\"" + "RelyingParty" + "\" : \"" + authenticationData.getRelyingParty()
+                + "\",\"" + "AuthenticatedIdPs" + "\" : \"" + authenticationData.getAuthenticatedIdps()
+                + "\"";
+
+        AUDIT_LOG.info(String.format(
+                FrameworkConstants.AUDIT_MESSAGE,
+                authenticationData.getAuthenticatedUser(),
+                "Login",
+                "ApplicationAuthenticationFramework", auditData, FrameworkConstants.AUDIT_SUCCESS));
+    }
+
+    protected void doPublishAuthenticationFailure(AuthenticationAuditData authenticationData) {
+
+        String auditData = "\"" + "ContextIdentifier" + "\" : \"" + authenticationData.getContextIdentifier()
+                + "\",\"" + "ServiceProviderName" + "\" : \"" + authenticationData.getServiceProvider()
+                + "\",\"" + "RequestType" + "\" : \"" + authenticationData.getInboundProtocol()
+                + "\",\"" + "RelyingParty" + "\" : \"" + authenticationData.getRelyingParty()
+                + "\",\"" + "StepNo" + "\" : \"" + authenticationData.getStepNo()
+                + "\"";
+
+        AUDIT_LOG.info(String.format(
+                FrameworkConstants.AUDIT_MESSAGE,
+                null,
+                "Login",
+                "ApplicationAuthenticationFramework", auditData, FrameworkConstants.AUDIT_FAILED));
+    }
+
+
+    protected void doPublishSessionTermination(AuthenticationContext context, String username,
+                                               String tenantDomain, String authenticatedIDPs) {
+
+        String auditData = "\"" + "ContextIdentifier" + "\" : \"" + context.getContextIdentifier()
+
+                + "\",\"" + "LoggedOutUser" + "\" : \"" + username
+                + "\",\"" + "LoggedOutUserTenantDomain" + "\" : \"" + tenantDomain
+                + "\",\"" + "ServiceProviderName" + "\" : \"" + context.getServiceProviderName()
+                + "\",\"" + "RequestType" + "\" : \"" + context.getRequestType()
+                + "\",\"" + "RelyingParty" + "\" : \"" + context.getRelyingParty()
+                + "\",\"" + "AuthenticatedIdPs" + "\" : \"" + authenticatedIDPs
+                + "\"";
+
+        String idpName = null;
+        ExternalIdPConfig externalIdPConfig = context.getExternalIdP();
+        if (externalIdPConfig != null) {
+            idpName = externalIdPConfig.getName();
+        }
+        AUDIT_LOG.info(String.format(
+                FrameworkConstants.AUDIT_MESSAGE,
+                username,
+                "Logout", idpName, auditData, FrameworkConstants.AUDIT_SUCCESS));
+    }
+
+
+
+    protected void publishSessionTermination(Event event) {
```

Comment:
Let's rename these methods to more meaningful names. Now we are no longer publishing anything and this is not extended from publishers. These handlers are writing to audit logs. So lets use more meaningful names.